Sources/MembraneRTC/Media/SoundDetection.swift

```diff
@@ -0,0 +1,179 @@
+import AVFoundation
+
+// Class for sound detection using the device's microphone
+public class SoundDetection: ObservableObject {
+
+  // Audio engine for sound detection
+  private var engine: AVAudioEngine!
+  private var inputNode: AVAudioInputNode!
+  private var mixerNode: AVAudioMixerNode!
+  private var onSoundDetectedListener: ((_ detectionResult: Bool) -> Void)?
+  private var onVolumeChangedListener: ((_ volume: Int) -> Void)?
+
+  // Property to indicate the recording state
+  @Published public var isRecording: Bool = false
+
+  /// Sets a listener to receive sound volume change events.
+  ///
+  /// - Parameter listener: The listener to be notified when sound volume changes.
+  public func setOnVolumeChangedListener(listener: ((_ volume: Int) -> Void)?) {
+    onVolumeChangedListener = listener
+  }
+
+  /// Sets a listener to receive sound detection events.
+  ///
+  /// - Parameter listener: The listener to be notified when a sound is detected.
+  public func setOnSoundDetectedListener(listener: ((_ detectionResult: Bool) -> Void)?) {
+    onSoundDetectedListener = listener
+  }
+
+  /// Initializer which sets up the audio session and audio engine.
+  public init() {
+    do {
+      try setupSession()
+      try setupEngine()
+    } catch let error {
+      print("Failed during initial setup: \(error.localizedDescription)")
```

Comment:
don't we want to throw exception for the user? how is it on Android, I remember we had similar discussion?